src/mpr/modules/utility/preParamsLookUp.ts

```diff
@@ -0,0 +1,1317 @@
+/**
+ * Technique Parameter
+ */
+export interface Param {
+  name: string;
+  mprReadType: 'Uint8' | 'Uint16' | 'Uint32' | 'Float32' | 'Float64' | 'string';
+  //to unit extracts from name
+  optional: boolean;
+}
+interface TechniquesToParams {
+  [techniqueName: string]: Param[];
+}
+/**
+ * Parameters ordered in the way they appear in the binary file,
+ * `optional` is to help the text parsing.
+ * `name` should be the name in the MPT file.
+ * The text-only parameter `Ns` appended at parsing time (it does not appear in the
+ * binary file.)
+ * For description of parameters see the manuals
+ * in the GH repo at `/docs`.
+ */
+export const preParamsLookUp: TechniquesToParams = {
```

Comment:
How did you generate this lookup?

Reply:
The first version of the MPR parser was not written by me, and this mapping was there already, you may want to check the first versions. 

I added the `textReadType` which was inferred from the test files, and now removed it because it does not really add much information.

I presume the original code was almost copy&paste from other code, and have since then got written of all the original code that could be inferred in an original manner. I've said this several times already (not to you I think.)

Reply:
@lpatiny I think you're the only other person who wrote code in this project. Do you remember how you created the mapping?

Reply:
I don't think luc wrote it, there are more contributors in the main page of the parser @targos

Reply:
> @lpatiny I think you're the only other person who wrote code in this project. Do you remember how you created the mapping?

[For example, this commit](https://github.com/cheminfo/biologic-converter/commit/8a349d12d16b68a983b2aec83d051368d04d8bec) is where MPR was added.